interface.go

```diff
@@ -20,3 +21,34 @@ type Session interface {
 	RemoteAddr() net.Addr
 	Close(error) error
 }
+
+// ConnState is the status of the connection
+type ConnState int
+
+const (
+	// ConnStateVersionNegotiated means that version negotiation is complete
+	ConnStateVersionNegotiated ConnState = iota
+	// ConnStateSecure means that the connection is encrypted
+	ConnStateSecure
+	// ConnStateForwardSecure means that the connection is forward secure
+	ConnStateForwardSecure
```

Comment:
I think the server-side is pretty clear here (`ConnStateVersionNegotiated` and `ConnStateSecure` happen at the same time, after receiving the first packet from the client. So I guess your comment is referring to the client-side of a 0-RTT.

In that case `ConnStateSecure` would happen immediately (i.e. before the first packet is sent), whereas `ConnStateVersionNegotiated` only fires after receiving the first packet from the server. This will require some changes in the client (I'm assuming an ordering of the `ConnState`s at some places), but I don't think any API change is needed.
In case of a rejected 0-RTT `ConnStateSecure` would fire twice though. This is something we probably should document sometime somewhere.